Sources/VLCActionSheet.swift

```diff
@@ -0,0 +1,422 @@
+    fileprivate func setupBottomBackgroundView() {
+        NSLayoutConstraint.activate([
+            bottomBackgroundViewHeightConstraint,
+            bottomBackgroundView.widthAnchor.constraint(equalTo: view.widthAnchor)
+            ])
+    }
+
+    fileprivate func setupMainStackViewConstraints() {
+        NSLayoutConstraint.activate([
+            mainStackView.bottomAnchor.constraint(equalTo: view.bottomAnchor),
+            mainStackView.leadingAnchor.constraint(equalTo: view.leadingAnchor),
+            mainStackView.widthAnchor.constraint(equalTo: view.widthAnchor),
+            mainStackView.trailingAnchor.constraint(equalTo: view.trailingAnchor)
+            ])
+    }
+
+    override func updateViewConstraints() {
+        super.updateViewConstraints()
+
+        if let presentingViewController = presentingViewController, let dataSource = dataSource {
+            collectionViewHeightConstraint.constant = CGFloat(dataSource.numberOfRows()) * cellHeight
+            maxCollectionViewHeightConstraint.constant = presentingViewController.view.frame.size.height / 2
+            collectionView.setNeedsLayout()
+            collectionView.layoutIfNeeded()
+        }
+    }
+
+    @available(iOS 11.0, *)
+    override func viewSafeAreaInsetsDidChange() {
+        bottomBackgroundViewHeightConstraint.constant = view.safeAreaInsets.bottom
+    }
+
+    // MARK: UIViewController
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        view.addSubview(backgroundView)
+        view.addSubview(mainStackView)
+
+        // wip: Currently adapting to new UI
+        //       option to hide cancelbutton
+        mainStackView.addArrangedSubview(headerView)
+        mainStackView.addArrangedSubview(collectionView)
+//        mainStackView.addArrangedSubview(cancelButton)
+        mainStackView.addArrangedSubview(bottomBackgroundView)
+
+        backgroundView.frame = UIScreen.main.bounds
+
+        setupMainStackViewConstraints()
+        setupCollectionViewConstraints()
+//        setupCancelButtonConstraints()
+        setuplHeaderViewConstraints()
+        setupBottomBackgroundView()
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        mainStackView.isHidden = true
+        updateViewConstraints()
+    }
+
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+
+        // This is to avoid a horrible visual glitch!
+        mainStackView.isHidden = false
+
+        UIView.transition(with: backgroundView, duration: 0.2, options: .transitionCrossDissolve, animations: { [weak self] in
+            self?.backgroundView.isHidden = false
+            }, completion: nil)
+
+        let realMainStackView = mainStackView.frame
+
+        mainStackView.frame.origin.y += mainStackView.frame.origin.y
+
+        UIView.animate(withDuration: 0.6, delay: 0, usingSpringWithDamping: 0.8, initialSpringVelocity: 0, options: .curveEaseOut, animations: {
```

Comment:
Does it make sense to share this animation in a global way with more animated objects? See the shared animation in the tvOS interface.

Reply:
I'm not sure, I don't think that it's that of a fancy animation that it deserves to be global?

Reply:
also why 0.6 ?

Reply:
0.3 is default and should be used could you otherwise share the animation with a gif / video to show what is actually going on here ?

Reply:
Well there is not much to see, `0.6` is only for the duration of the animation. I felt that 0.3 was way to fast.